pkg/operator/controller/certificate/default_cert.go

```diff
@@ -17,67 +17,77 @@ import (
 
 // ensureDefaultCertificateForIngress creates or deletes an operator-generated
 // default certificate for a given IngressController as appropriate.  Returns true
-// if it creates, updates, or deletes the secret for the certificate, false
-// otherwise.
+// if it the secret exists, or false if it does not, as well as any errors.
 func (r *reconciler) ensureDefaultCertificateForIngress(caSecret *corev1.Secret, namespace string, deploymentRef metav1.OwnerReference, ci *operatorv1.IngressController) (bool, error) {
 	ca, err := crypto.GetCAFromBytes(caSecret.Data["tls.crt"], caSecret.Data["tls.key"])
 	if err != nil {
 		return false, fmt.Errorf("failed to get CA from secret %s/%s: %v", caSecret.Namespace, caSecret.Name, err)
 	}
-	desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
+	wantCert, desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
 	if err != nil {
 		return false, err
 	}
-	current, err := r.currentRouterDefaultCertificate(ci, namespace)
+	if !wantCert {
+		// If the operator generated certificate is not being used, ensure that the ingress controller's
+		// Spec.DefaultCertificate secret exists before deleting the operator generated secret.
+		// See https://bugzilla.redhat.com/show_bug.cgi?id=1887441
+		err := r.lookupUserSpecifiedRouterDefaultCertificate(ci, namespace)
+		if err != nil {
+			return false, fmt.Errorf("failed to lookup user specified default certificate: %v", err)
+		}
+	}
+
+	haveCert, current, err := r.currentRouterDefaultCertificate(ci, namespace)
```

Comment:
Is it important to verify that the current cert secret also exists?  What if the `currentRouterDefaultCertificate` doesn't exist, was accidentally (or purposely) deleted?

Reply:
`currentRouterDefaultCertificate` already does this and we handle any error/false condition from it, right?

Reply:
Looks like if it's not found then it returns false `haveCert` but nil `err`:
https://github.com/openshift/cluster-ingress-operator/pull/505/files#diff-16d9c01d5eac10bd8a3012d74ef5feb542287746e676beca44479eb80cbaa124R116

We don't handle/report any error on the inability to locate the current default certificate.  Then it just overwrites it with the existing default.  If I understand this properly.

Reply:
If we are unable to locate the current operator generated default certificate, but we want the operator to generate one, then we create it. In the case where `haveCert` and `wantCert` are true, we no-op, basically.

Reply:
I think there's a special case if we're unable to find the secret due to a timeout on the api.   Wdyt?

Reply:
> I think there's a special case if we're unable to find the secret due to a timeout on the api. Wdyt?

In the case, `err != nil` on the next line would evaluate to true, `ensureDefaultCertificateForIngress` would return the error early, and the reconcile loop would retry the ensure again.